src/main/java/seedu/clinic/logic/commands/DeleteCommand.java

```diff
@@ -21,59 +34,133 @@
+            return executeSupplierRelatedDeletion(model);
         } else {
-            List<Warehouse> lastShownList = model.getFilteredWarehouseList();
+            return executeWarehouseRelatedDeletion(model);
+        }
+    }
 
-            if (targetIndex.getZeroBased() >= lastShownList.size()) {
-                throw new CommandException(Messages.MESSAGE_INVALID_WAREHOUSE_DISPLAYED_INDEX);
-            }
+    private CommandResult executeWarehouseRelatedDeletion(Model model) throws CommandException {
+        List<Warehouse> lastShownList = model.getFilteredWarehouseList();
 
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_WAREHOUSE_DISPLAYED_INDEX);
+        }
+
+        if (targetType.equals(WAREHOUSE)) {
             Warehouse warehouseToDelete = lastShownList.get(targetIndex.getZeroBased());
             model.deleteWarehouse(warehouseToDelete);
             return new CommandResult(String.format(MESSAGE_DELETE_WAREHOUSE_SUCCESS, warehouseToDelete));
         }
+
+        Warehouse warehouseToUpdate = lastShownList.get(targetIndex.getZeroBased());
+        Set<Product> productSetToUpdate = new HashSet<>(warehouseToUpdate.getProducts());
+        Product matchedProduct;
+        try {
+            matchedProduct = warehouseToUpdate.getProductByName(targetProductName.get());
+            productSetToUpdate.remove(matchedProduct); // removes the matching product
+            Warehouse updatedWarehouse = new Warehouse(warehouseToUpdate.getName(), warehouseToUpdate.getPhone(),
+                    warehouseToUpdate.getAddress(), warehouseToUpdate.getRemark(), productSetToUpdate);
+            model.setWarehouse(warehouseToUpdate, updatedWarehouse);
+            model.updateFilteredWarehouseList(PREDICATE_SHOW_ALL_WAREHOUSES);
+            return new CommandResult(String.format(MESSAGE_DELETE_PRODUCT_IN_WAREHOUSE_SUCCESS,
+                    matchedProduct.getProductName(), updatedWarehouse.getName()));
+        } catch (ProductNotFoundException e) {
+            throw new CommandException(String.format(Messages.MESSAGE_INVALID_WAREHOUSE_DISPLAYED_INDEX,
+                    targetProductName.get(), warehouseToUpdate.getName()));
+        }
+    }
+
+    private CommandResult executeSupplierRelatedDeletion(Model model) throws CommandException {
+        List<Supplier> lastShownList = model.getFilteredSupplierList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_SUPPLIER_DISPLAYED_INDEX);
+        }
+
+        if (targetType.equals(SUPPLIER)) {
+            Supplier supplierToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deleteSupplier(supplierToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_SUPPLIER_SUCCESS, supplierToDelete));
+        }
```

Comment:
Just wish to clarify, is there a need for the if clause here since there is a conditional check above (lines 89-90) where this method would be called if targetType.equals(SUPPLIER). Could there potentially be any cases where targetType is not equals to SUPPLIER and this method could be invoked?

Reply:
Actually for the execute method, we first classify into supplierRelated and warehouseRelated. Afterwards in the supplierRelated executions, we need to split into supplier deletion and product for supplier deletion. That's the purpose of this if statement here.

Reply:
Ah okay, that explains! :) All is clear now, no issues with this part 👍